src/dotty/tools/dotc/ast/Trees.scala

```diff
@@ -317,6 +317,10 @@ object Trees {
     private[ast] def rawMods: Modifiers[T] =
       if (myMods == null) genericEmptyModifiers else myMods
 
+    private[this] var myComment: Option[String] = None
```

Comment:
hmmm, adding new fields in `trees` is likely to increase memory footprint.
AS most trees do not have comments,@felixmulder, would you consider putting comments into attachments of the tree?